bindings/qt/sdk.pri

```diff
@@ -58,9 +58,8 @@ CONFIG(USE_MEGAAPI) {
   }
 }
 
-CONFIG += c++11
 !win32 {
-    QMAKE_CXXFLAGS += -std=c++11
+    QMAKE_CXXFLAGS += -std=c++11 -Wextra -Wconversion -Wno-unused-parameter
```

Comment:
Yes and we still are. On a different line I simply removed `CONFIG += c++11` which ensures we're not using gnu extensions.
For a future change: We shouldn't actually define the language standard here at all because it affects downstream projects. Like if a project using the SDK decided to use C++17 it simply can't because we're overriding it here with C++11.

Reply:
I disagree. The SDK uses an specific version of the C standard. As today, it is C++11, so the main `.pri` file should specify it. Everything in the SDK must be able to compile using C++11. If a sub-project like the sync-tests want to use a different version, it should only apply to that project and you can unset the `c++11` and set the `c++17`
```
CONFIG -= c++11
QMAKE_CXXFLAGS-=-std=c++11
CONFIG += c++17
QMAKE_CXXFLAGS+=-std=c++17
```

Reply:
That seems a little cumbersome, no? IMO the build target should define the standard. Of course, the SDK will only compile with at least C++11 but also with anything newer.

Reply:
Note that the SDK itself is a target as well for the apps, who use the `sdk.pri` for that purpose, so according to your idea it should define the standard too :)
A suggestion from @polmr is to define a flag to allow SDK's target apps to decide. Imagine the following in the `sdk.pri`:
```
!CONFIG(DO_NOT_SET_CPP_STANDARD) 
{
    CONFIG+=C++11
}
```
and in the SDK's target apps, their `.pro`s may define their desired standard before including the `sdk.pri` by:
```
CONFIG+=C++17
CONFIG+=DO_NOT_SET_CPP_STANDARD
```
That way, we have the best of both worlds. How does it sound?

Asi dejamos evolucionar a ambar partes por separado: podemos pasar a usar c++14 en el sdk.pri sin tener que tocar configuracion de las apps, y tenemos un standar usado por defecto en caso de que no este establecido por la aplicacion

Reply:
Yeah that seems reasonable. The only problem with `CONFIG+=C++11` is that QMake, for whatever reason, adds gnu extensions which it really shouldn't. 
Sorry I wasn't clear what I meant by "build target". I meant an executable target should define the standard. However, configuring it the way you suggested should work too :)

Reply:
in CMAKE you can iterate on the items in the CXX_FLAGS variables, and check if any standard was set, and add -std=c++11 if none was set by the enclosing application.  Maybe the same approach is possible in qmake?   Just a thought

Reply:
I don't know if `CXX_FLAGS` can be checked, but if you feel the `DO_NOT_SET_CPP_STANDARD` makes the trick, go ahead, please, and push the change :)